server/src/main/java/com/decathlon/ara/defect/DefectAdapter.java

```diff
@@ -63,15 +63,15 @@
      * @param id        a user-typed defect ID
      * @return true if it is a valid ID for the backed defect tracking system
      */
-    boolean isValidId(long projectId, String id);
+    Boolean isValidId(Long projectId, String id);
```

Comment:
Hello, how are you?

Why the change from primitive non-nullables to nullable-objects?

The goal of having primitive types was to document the fact that parameters cannot be null, and that the method effectively return a two-state boolean (true/false), and not a tri-state boolean (true/false/null).

This non-null information was of great value when arriving on a new source file: it avoids maintainers to have to think if null is an allowed value, and what to do when there is a null value: avoid developers to write a lot of defencive code to handle null when in fact it will never happen, and have no business meaning.

Reply:
thank you for your feed back !